ui-tests/test/notebook.spec.ts

```diff
@@ -91,4 +91,47 @@ test.describe('Notebook', () => {
     // check the short output area is not auto scrolled
     expect(await checkCell(1)).toBe(false);
   });
+
+  test('Open table of content left panel', async ({ page, tmpPath }) => {
+    const notebook = 'simple_toc.ipynb';
+    const menuPath = 'View>Show Left Sidebar>Table Of Contents';
+    await page.contents.uploadFile(
+      path.resolve(__dirname, `./notebooks/${notebook}`),
+      `${tmpPath}/${notebook}`
+    );
+    await page.goto(`notebooks/${tmpPath}/${notebook}`);
+
+    await waitForKernelReady(page);
+
+    await page.menu.open(menuPath);
+
+    await expect(
+      page.locator(
+        '.jp-SidePanel-content > .jp-TableOfContents-tree > .jp-TableOfContents-content'
+      )
+    ).toHaveCount(1);
+    await expect(
+      page.locator(
+        '.jp-SidePanel-content > .jp-TableOfContents-tree > .jp-TableOfContents-content > .jp-tocItem'
+      )
+    ).toHaveCount(3);
+  });
+
+  test('Open notebook tools right panel', async ({ page, tmpPath }) => {
+    const notebook = 'simple.ipynb';
+    const menuPath = 'View>Show Right Sidebar>Table Of Contents';
+    await page.contents.uploadFile(
+      path.resolve(__dirname, `./notebooks/${notebook}`),
+      `${tmpPath}/${notebook}`
+    );
+    await page.goto(`notebooks/${tmpPath}/${notebook}`);
+
+    await waitForKernelReady(page);
+
+    await page.menu.open(menuPath);
+
+    await page.isVisible('#notebook-tools.jp-NotebookTools');
+
+    await page.isVisible('#notebook-tools.jp-NotebookTools > #add-tag.tag');
```

Comment:
Thanks for adding a test here this is going to be useful to catch potential regressions :+1: 

Wondering whether we should also add new reference snapshots to make sure the side panels render correctly?

Reply:
This could also be done in a follow-up PR.

Reply:
Added snapshots on *toc* and *NotebookTools* panels in https://github.com/jupyter/notebook/pull/6487/commits/7062af6cda1bbcab4d92d17f27a62db275596192